src/Database/Expression/TupleComparison.php

```diff
@@ -51,6 +51,16 @@ public function __construct($fields, $values, array $types = [], string $conjunc
         $this->_operator = $conjunction;
     }
 
+    /**
+     * Returns the type to be used for casting the value to a database representation
+     *
+     * @return array
```

Comment:
I think they are.

Reply:
If I were to pass expressions, it would feel weird to having to declare a type if it's not actually used... If no one has a good reason for why `null` shouldn't be allowed, I'll revert this, I don't want to blow up peoples CI for no reason.

Reply:
The current reason to only use phpstan/psalm tags for this and keep default tags normal (legacy), is that these kind of syntax updates are not yet known to many IDEs.
Or at least this was the case in the last years.
If the IDEs now caught up, we can lose the restriction around having to use the p*- tags on top and can directly modify the tags itself with new syntax usage.